File: Binary_Tree/94-m-inorderTraversal.cpp
// Definition for a binary tree node.
#include <vector> 

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};
 
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> v;
        inTraversal(root, v);
        return v;
    }
    void inTraversal(TreeNode* root, vector<int>& v) {
        if (!root)
            return;
        inTraversal(root->left, v);
        v.push_back(root->val);
        inTraversal(root->right, v);
    }
};

File: Binary_Tree/94-m-inorderTraversal.py
# Definition for a binary tree node.
class TreeNode:
     def __init__(self, x):
         self.val = x
         self.left = None
         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        res,temp=[],[]
        while 1:
            while root:
                temp.append(root)
                root=root.left
            if not temp:return res
            root=temp.pop()
            res.append(root.val)
            root=root.right